Cabal/Distribution/Backpack/ModuleScope.hs

```diff
@@ -58,29 +57,28 @@ import qualified Data.Set as Set
 -- can influence the 'ModuleShape' via a reexport.
 data ModuleScope = ModuleScope {
     modScopeProvides :: ModuleProvides,
-    modScopeRequires :: Set ModuleName
+    modScopeRequires :: ModuleRequires
     }
 
 -- | Every 'Module' in scope at a 'ModuleName' is annotated with
 -- the 'PackageName' it comes from.
 type ModuleProvides = Map ModuleName [ModuleSource]
+-- | INVARIANT: entries for ModuleName m, have msrc_module is OpenModuleVar m
+type ModuleRequires = Map ModuleName [ModuleSource]
```

Comment:
Nice. Maybe even better to have a newtype, perhaps in a follow-up to make both types newtypes? I realize that such newtypes aren't exactly idiomatic in Cabal :).